tests/python_package_test/utils.py

```diff
@@ -0,0 +1,15 @@
+
+try:
+    from functools import lru_cache
+except ImportError:
+    import warnings
+    warnings.warn("Could not import functools.lru_cache", RuntimeWarning)
+
+    def lru_cache(user_function, maxsize=None):
+        @wraps(user_function)
```

Comment:
You should import `wraps` I think: https://docs.python.org/2.7/library/functools.html#functools.wraps.

Reply:
😆 ok it's time for me to just make a python 2.7 conda env and actually test this before pushing commits

Reply:
ok that worked! I didn't install `lightgbm` in that env, but I was able to test the data loading

```
conda create --name py27 python=2.7 scikit-learn
```

and it looks like all tests are passing now